### ISSUE TITLE
Support silent install, no frame or terminal

### ISSUE THREAD

**Issue by perfectspr** [kkoomen/vim-doge#199]
vim-doge will block install progress in silent mode:

```bash
vim +PlugInstall +qall > /dev/null
```

I'm packing vim into a Docker image and use vim-plug to manage all plugins.

**Comment by kkoomen** [comment]
@subnut What do you think? Should this command wait for the install to be done or not? To me, the behavior is as expected. You want to be sure that the binary is available, *but* if `PlugInstall` will run in the background then so should we. Can we implement this easily?

**Comment by subnut** [comment]
Can we detect silent mode? If yes, how?

**Comment by kkoomen** [comment]
@subnut Well, how does vim-plug do it? I know they run background processes, but I am not sure if their way already covers this scenario, or whether they have implement an additional check that we might need as well.

**Comment by subnut** [comment]
> Well, how does vim-plug do it?

@kkoomen I really don't know!

---

@perfectspr Check if this works -

```diff
diff --git a/autoload/doge.vim b/autoload/doge.vim
index 30df14a..f736688 100644
--- a/autoload/doge.vim
+++ b/autoload/doge.vim
@@ -180,7 +180,7 @@ endfunction
 
 "" @public
 " Install the necessary dependencies.
-function! doge#install() abort
+function! doge#install(...) abort
   for l:filename in ['vim-doge', 'vim-doge.exe']
     let l:filepath = g:doge_dir . '/bin/' . l:filename
     if filereadable(l:filepath)
@@ -212,6 +212,11 @@ function! doge#install() abort
     let l:term_height = 4
   endif
 
+  if len(a:000) > 0 && a:0
+    call system(l:command)
+    return
+  endif
+
   if has('nvim') && exists(':terminal') == 2
     " neovim with :terminal support
     " vint: next-line -ProhibitUnusedVariable
```

```
vim +PlugUpdate '+call doge#install(1)' +qall > /dev/null
```

**Comment by kkoomen** [comment]
@subnut If it works, then perhaps add `silent` in front of the `call system()` command.

**Comment by subnut** [comment]
> add `silent` in front of the `call system()` command.

Umm.... Why?

**Comment by perfectspr** [comment]
It works! thanks for your quick implementation @subnut

**Comment by kkoomen** [comment]
@subnut Nice! I think we can name it something like `{'background': 1}` to indicate you want to return it in the background.

**Comment by subnut** [comment]
But that wouldn't imply that it will block/freeze there UI

**Comment by subnut** [comment]
`script_mode` maybe?

**Comment by kkoomen** [comment]
We can do `ci` or perhaps `headless`. Both indicate the action that OP wants.

**Comment by subnut** [comment]
I agree. Let's go with `headless`.

**Comment by kkoomen** [comment]
@subnut No problem. You're helping out so I'll definitely do that ;)

**Comment by kkoomen** [comment]
@subnut Can I use `Subhaditya Nath <sn03.general@gmail.com>` as the `Co-authored-by`?

**Comment by subnut** [comment]
> @subnut Can I use `Subhaditya Nath <sn03.general@gmail.com>` as the `Co-authored-by`?

@kkoomen Yes. Thanks again.

**Comment by kkoomen** [comment]
This feature has been merged and released in [v3.6.0](https://github.com/kkoomen/vim-doge/tree/v3.6.0).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.

**Comment by kkoomen** [comment]
@perfectspr Please read the new [Headless mode](https://github.com/kkoomen/vim-doge#headless-mode) section for more details.
